### PR TITLE
Mock Pipeline

### PR DESCRIPTION
**Pull request by NinadBhat**
Issue: #314

### CODE REVIEW

File: src/shipit_pipeline/shipit_pipeline/api.yml [hunk 1]
```diff
@@ -65,6 +65,31 @@ paths:
         200:
           description: Removal of pipeline
 
+  # Mock Step
+  /mock/{uid}:
+    get:
+      operationId: "shipit_pipeline.pipeline.get_step"
```
Comment (garbas) [inline src/shipit_pipeline/shipit_pipeline/api.yml]: change ``operationId`` to ``"shipit_pipeline.api.get_mock"``

File: src/shipit_pipeline/shipit_pipeline/api.yml [hunk 2]
```diff
@@ -65,6 +65,31 @@ paths:
...
+          - $ref: '#/parameters/uid'
+      responses:
+        200:
+          description: Definition of Step
+          schema:
+            $ref: '#/definitions/Step'
+
+  /mock/{uid}/status:
+    get:
+      operationId: "shipit_pipeline.pipeline.get_state"
```
Comment (garbas) [inline src/shipit_pipeline/shipit_pipeline/api.yml]: change ``operationId`` to ``"shipit_pipeline.api.get_mock_status"``

File: src/shipit_pipeline/shipit_pipeline/pipeline.py [hunk 3]
```diff
@@ -52,6 +53,77 @@ def is_pending(self):
...
+    step = MockPipelineStep(uid, params, requires, state=state, step_time=step_time)
+    MockSteps[uid] = step
+
+def start_Step(uid):
+    return MockSteps[uid].start_step()
+
+def get_state(uid):
+    return MockSteps[uid].get_state()
+
+def get_step(uid):
```
Comment (garbas) [inline src/shipit_pipeline/shipit_pipeline/pipeline.py]: Move this method to ``api.py`` and rename it to ``get_mock``

File: src/shipit_pipeline/shipit_pipeline/pipeline.py [hunk 4]
```diff
@@ -52,6 +53,77 @@ def is_pending(self):
...
+
+
+def add_step(uid, params, requires, state='pending', step_time=100):
+    step = MockPipelineStep(uid, params, requires, state=state, step_time=step_time)
+    MockSteps[uid] = step
+
+def start_Step(uid):
+    return MockSteps[uid].start_step()
+
+def get_state(uid):
```
Comment (garbas) [inline src/shipit_pipeline/shipit_pipeline/pipeline.py]: Move this method to ``api.py`` and rename it to ``get_mock_status``

File: src/shipit_pipeline/shipit_pipeline/pipeline.py (hunk 3 above)
Comment (NinadBhat) [inline src/shipit_pipeline/shipit_pipeline/pipeline.py]: Will do this tomorrow.

**Comment by garbas** [comment]
i just rebased your work on current master (and resolve the conflicts). I'm waiting now for taskcluster to finish to see what we need to fix next.

**Comment by garbas** [comment]
@NinadBhat looks like flake8 errors happen. to be able to reproduce this errors run:

```console
~/services $ ./please shell shipit-pipeline
(nix-shell) ~/services/src/shipit_pipeline $ flake8
```

**Comment by garbas** [comment]
@NinadBhat this PR is not WIP anymore, right?

**Comment by NinadBhat** [comment]
@garbas : It is no longer WIP. You can review it.

**Pull request merged**